### ISSUE TITLE
Support --nohoist option for workspaces.

### ISSUE THREAD

**Issue by richburdon** [yarnpkg/yarn#4049]
Feature:

Workspaces currently hoist everything to the root level. `lerna` supports a `--nohoist` option to leave modules that match a regex in the `packages/sub/node_modules` folder.

For example, `grunt` requires that all plugins are in the local `node_modules` directory.

**Comment by MoOx** [comment]
I have this problem when using reasonml packages.
Some of my packages of my workspace require bs-platform + some others other of my own, as deps, but with everything at the root, `bsb` is throwing some fatal errors like (since you have bsb-platform at the root of the project): `Error: Unbound module Pervasives` (bs-platform does not like to be relocated).
I tried to add older version at the root of the project so the right version is in the right place for my package and I then got `BSB check build spec : Dependencies information missing` because some deps are not at the right place for `bsb`. Using a nohoist glob or something like lerna will make it usable for reasonml packages.
